File: Data Structures & Algorithms/binary-search/submission-9.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
      let l = 0;
      let r = nums.length - 1;

      while (l<=r) {
        let m = Math.floor((l + r) / 2);

        if (target === nums[m]) {
            return m;
        } else if (nums[m] < target) {
            l = m + 1;
        } else {
            r = m - 1;
        }
      } 
      return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-9.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l, r = 0, len(nums) - 1

        while l < r:
            two_sum = nums[l] + nums[r]
            if two_sum == target:
                return [l+1, r+1]
            elif two_sum < target:
                l += 1
            else:
                r -= 1
        return "WHAT THE FUCK!"
        